### PR TITLE
fix(dsg): type error in entity-title

### PR DESCRIPTION
**Pull request by Shurtu-gal**
Close: #6896 

### Summary
🛠️🚀🎨

Improved the `ENTITY_TITLE` function to handle different field types in the data service generator. This function is used to render the title of an entity record in the `entity-title-component.template.tsx` file.

> _`ENTITY_TITLE`_
> _Adapts to field types - a change_
> _For low-code platform_

### Walkthrough
*  Handle non-string entity title fields in the admin UI ([link](https://github.com/amplication/amplication/pull/6897/files?diff=unified&w=0#diff-db55a9b3321c8b75fb2e8f2a61546a3e16e97f292cf19b5b8d2692ff2dd69cf6L10-R12))

## PR Checklist

- [x] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

File: packages/data-service-generator/src/admin/entity/entity-title-component/entity-title-component.template.tsx [hunk 1]
```diff
@@ -7,5 +7,7 @@ declare const ENTITY_TITLE_FIELD_NAME = "title";
 export const ENTITY_TITLE_FIELD_NAME_ID = ENTITY_TITLE_FIELD_NAME;
 
 export const ENTITY_TITLE = (record: ENTITY): string => {
-  return record.ENTITY_TITLE_FIELD || String(record.id);
+  return typeof record[ENTITY_TITLE_FIELD_NAME] === "string"
```
Comment (barshimi) [inline packages/data-service-generator/src/admin/entity/entity-title-component/entity-title-component.template.tsx:10]: you better change it to: Object.prototype.toString.call(str) === "[object String]"

Comment (Shurtu-gal) [reply packages/data-service-generator/src/admin/entity/entity-title-component/entity-title-component.template.tsx]: Done @barshimi.

**Comment by overbit** [comment]
@Shurtu-gal thanks for the PR, can you have a look at the failing test? 
Once that is fixed, I'll approve it and merge it

**Comment by Shurtu-gal** [comment]
@overbit I have fixed it.

**Review by abrl91 — APPROVED** [review]

**Comment by abrl91** [comment]
@Shurtu-gal the snapshots didn't change?

File: packages/data-service-generator/src/admin/entity/entity-title-component/entity-title-component.template.tsx [hunk 2]
```diff
@@ -7,5 +7,5 @@ declare const ENTITY_TITLE_FIELD_NAME = "title";
 export const ENTITY_TITLE_FIELD_NAME_ID = ENTITY_TITLE_FIELD_NAME;
 
 export const ENTITY_TITLE = (record: ENTITY): string => {
-  return record.ENTITY_TITLE_FIELD || String(record.id);
+  return record.ENTITY_TITLE_FIELD?.toString() || String(record.id);
```
Comment (overbit) [inline packages/data-service-generator/src/admin/entity/entity-title-component/entity-title-component.template.tsx:10]: in order to update the snapshot testing with the new expectation, you can run `npx nx update-snapshot data-service-generator` @Shurtu-gal

Comment (Shurtu-gal) [reply packages/data-service-generator/src/admin/entity/entity-title-component/entity-title-component.template.tsx:10]: Thanks

Comment (Shurtu-gal) [reply packages/data-service-generator/src/admin/entity/entity-title-component/entity-title-component.template.tsx:10]: Done

**Review by mulygottlieb — APPROVED** [review]
Nicely done @Shurtu-gal 🚀

File: packages/data-service-generator/src/admin/entity/entity-title-component/entity-title-component.template.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Shurtu-gal) [reply packages/data-service-generator/src/admin/entity/entity-title-component/entity-title-component.template.tsx:10]: Hey, I tried that and it gives String(record.id) || String(record.id) in Profile Title. Is that fine.

**Pull request merged**